Mage.Tests/src/test/java/org/mage/test/cards/single/clb/ScionOfHalasterTest.java

```diff
@@ -0,0 +1,81 @@
+package org.mage.test.cards.single.clb;
+
+import mage.constants.PhaseStep;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.GameException;
+import org.junit.Test;
+import org.mage.test.serverside.base.CardTestCommanderDuelBase;
+
+import java.io.FileNotFoundException;
+
+/**
+ *
+ * @author Rjayz
+ */
+public class ScionOfHalasterTest extends CardTestCommanderDuelBase {
+
+    @Override
+    protected Game createNewGameAndPlayers() throws GameException, FileNotFoundException {
+        setDecknamePlayerA("CommanderDuel_Partners.dck"); // Commander = Ishai, Ojutai Dragonspeaker and Thrasios, Triton Hero
+        return super.createNewGameAndPlayers();
+    }
+
+    @Test
+    public void TestScionOfHalaster() {
+        addCard(Zone.BATTLEFIELD, playerA, "Scion of Halaster", 1);
+
+        addCard(Zone.BATTLEFIELD, playerA, "Plains", 1);
+        addCard(Zone.BATTLEFIELD, playerA, "Island", 4);
+        addCard(Zone.BATTLEFIELD, playerA, "Forest", 1);
+
+        // Cast commander so Scion of Halaster takes effect
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Thrasios, Triton Hero");
+
+        playerA.getLibrary().clear();
+        addCard(Zone.LIBRARY, playerA, "Lightning Bolt");
+        addCard(Zone.LIBRARY, playerA, "Mountain");
+        skipInitShuffling();
+
+        // Continue to turn 3 so we get to Player A's draw step
+        setStopAt(3, PhaseStep.PRECOMBAT_MAIN);
+        execute();
+
+        // AI considers Lightning Bolt a better card than Mountain so it will discard the Mountain
```

Comment:
If you use `setStrictChooseMode(true)` you can manually set which choices it should make.